### PR TITLE
#5 Card, item, search

### PR DESCRIPTION
**Pull request by 3rdvision**
| - | - |
| --- | --- |
| Issue | https://github.com/ripe-tech/ripe-id-mobile/issues/5 |
| Decisions | - Add card component <br> - Small fix story storybook name |
| Animated GIF | image |

### CODE REVIEW

File: react/components/molecules/card/card.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
+import React, { PureComponent } from "react";
+import { StyleSheet, Text, View, ViewPropTypes } from "react-native";
+import PropTypes from "prop-types";
+
+import { Avatar, Icon, Touchable } from "../../atoms";
+
+import { baseStyles } from "../../../util";
```
Comment (BeeMargarida) [inline react/components/molecules/card/card.js:7]: Move this above atoms import. Normally we order by how "external" and "removed" it is to the current directory.

File: react/components/molecules/card/card.js [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+    _renderCardImage() {
+        if (this.props.icon) return this._renderCardImageIcon();
+        if (this.props.avatarURL) return this._renderCardImageAvatar();
+    }
+
+    _renderCardText() {
+        return (
+            <View style={styles.textContainer}>
+                {this.props.title ? <Text style={styles.title}>{this.props.title}</Text> : null}
```
Comment (BeeMargarida) [inline react/components/molecules/card/card.js:89]: Just to simplify, perhaps do:
```suggestion
                {this.props.title && <Text style={styles.title}>{this.props.title}</Text>}

```

File: react/components/molecules/card/card.js [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        );
+    }
+
+    render() {
+        return (
+            <Touchable
+                style={styles.style}
+                onPress={this.props.onPress}
+                onLongPress={this.props.onLongPress}
+                activeOpacity={0.6}
```
Comment (BeeMargarida) [inline react/components/molecules/card/card.js:102]: Move `activeOpacity` above `on` functions, since this is normally the standard we use

File: react/components/molecules/card/card.js [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+    static get propTypes() {
+        return {
+            avatarURL: PropTypes.String,
+            icon: PropTypes.String,
+            iconColor: PropTypes.String,
+            iconFill: PropTypes.String,
+            title: PropTypes.String,
+            text: PropTypes.String,
+            subText: PropTypes.String,
+            style: ViewPropTypes.style,
```
Comment (BeeMargarida) [inline react/components/molecules/card/card.js:19]: style is always the last prop (also in defaultProps)

File: react/components/molecules/card/card.stories.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
+import React from "react";
+import { storiesOf } from "@storybook/react-native";
+import { withKnobs, select, text } from "@storybook/addon-knobs";
+import { View } from "react-native";
```
Comment (BeeMargarida) [inline react/components/molecules/card/card.stories.js:4]: Nitpicking again, but move this below "react" and before "storybook" imports pls

File: react/components/molecules/radio-group/radio-group.stories.js [hunk 6]
```diff
@@ -4,7 +4,7 @@ import { withKnobs, boolean, select } from "@storybook/addon-knobs";
 
 import { RadioGroup } from "./radio-group";
 
-storiesOf("Components/Molecules/Radio Group", module)
+storiesOf("Molecules", module)
```
Comment (BeeMargarida) [inline react/components/molecules/radio-group/radio-group.stories.js:7]: This will have to be made in another PR

File: react/components/molecules/card/card.js [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+    contentStyle: {
+        alignItems: "center",
+        backgroundColor: "#ffffff",
+        flexDirection: "row",
+        height: 100,
+        shadowColor: "#435764",
+        elevation: 5,
+        shadowOffset: { width: 0, height: 2 },
+        shadowOpacity: 0.4,
+        overflow: "hidden",
```
Comment (BeeMargarida) [inline react/components/molecules/card/card.js:167]: This line is causing the shadow to not appear in iOS
image

**Comment by BeeMargarida** [comment]
Also, since the search component was already merged with this PR, there is some misalign in the icon in iOS:
image

**Comment by 3rdvision** [comment]
cold be something related to an outdated setup of some sort?

storybook:
image

**Review by BeeMargarida — APPROVED** [review]

File: react/components/molecules/card/card.js [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+                icon={this.props.icon}
+                color={this.props.iconColor}
+                fill={this.props.iconFill}
+                height={60}
+                width={60}
+            />
+        );
+    }
+
+    _renderCardImage() {
```
Comment (gcandal) [inline react/components/molecules/card/card.js:103]: return explicit null

File: react/components/molecules/card/card.js [hunk 9]
```diff
@@ -0,0 +1,198 @@
...
+        if (this.props.icon) return this._renderCardImageIcon();
+        if (this.props.avatarURL) return this._renderCardImageAvatar();
+    }
+
+    _renderCardText() {
+        return (
+            <View style={styles.textContainer}>
```
Comment (gcandal) [inline react/components/molecules/card/card.js:110]: Don't return view if there isn't any of the 3 props

File: react/components/molecules/search/search.js [hunk 10]
```diff
@@ -0,0 +1,148 @@
+import React, { PureComponent } from "react";
```
Comment (gcandal) [inline react/components/molecules/search/search.js:1]: Merge with master to get rid of this from the diff

Comment (3rdvision) [reply react/components/molecules/search/search.js:1]: search is not in master but it was already accepted and merged here

**Review by gcandal — APPROVED** [review]

**Review by joamag — APPROVED** [review]
